mud-contracts/world/src/modules/kill-mail/systems/KillMail.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { EveSystem } from "@eveworld/smart-object-framework/src/systems/internal/EveSystem.sol";
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { KillMailLossType } from "../../../codegen/common.sol";
+import { KillMailTable } from "../../../codegen/tables/KillMailTable.sol";
+import { IKillMailErrors } from "../IKillMailErrors.sol";
+import { Utils } from "../Utils.sol";
+
+contract KillMail is EveSystem {
+  function reportKill(
+    uint256 killMailId,
+    address killer,
+    address victim,
+    KillMailLossType lossType,
+    uint256 killTimestamp
+  ) public hookable(killMailId, _systemId()) {
+    ResourceId tableId = Utils.killMailTableId(_namespace());
+
+    if (KillMailTable.getKiller(tableId, killMailId) != address(0)) {
```

Comment:
we're getting the whole kill 
```suggestion
    if (KillMailTable.getKill(tableId, killMailId) != address(0)) {
```

Reply:
This is just to check if a kill has been registered or not.

Reply:
Oh I was confused, so shouldn't it be:
```suggestion
    if (KillMailTable.getKiller(tableId, killMailId) == address(0)) {
```

Or maybe I'm missing something

Reply:
oh `address(0)` is not the `killer` argument 🤦 nevermind, all good